xdsl/passes.py

```diff
@@ -103,6 +104,35 @@ def from_pass_spec(cls: type[ModulePassT], spec: PipelinePassSpec) -> ModulePass
         return cls(**arg_dict)
 
 
+def _empty_callback(
+    previous_pass: ModulePass, module: builtin.ModuleOp, next_pass: ModulePass
+) -> None:
+    return
+
+
+@dataclass
+class PassPipelinePass(ModulePass):
+    passes: list[ModulePass]
+    callback: Callable[[ModulePass, builtin.ModuleOp, ModulePass], None] = field(
+        default=_empty_callback
+    )
+    """
+    Function called in between every pass, taking the pass that just ran, the module, and
+    the next pass.
```

Comment:
Should this be run after every pass, or in between passes?
Should we have two options? Not sure. 
I feel this kind of complexity may not make this worth it?

Reply:
I think in between passes is the only thing that this class needs to provide, the client can call their own code before or after executing the pipeline

Reply:
After executing the pipeline, a lot of state will be deleted, right? So you might want to have several options of callbacks 🤔

Reply:
Not sure exactly what you mean